### ISSUE TITLE
Refactor recursive calls in sourcers

### ISSUE THREAD

**Issue by mmaruniak** [Cimpress-MCP/JustReleaseNotes#48]
As the default maximum limit of recursive calls in Python is set to 1000, we may experience issues when processing commits of bigger repositories. The current workaround is to artificially increase this stack size. It would be nicer to avoid modifying system settings and refactor such calls into loops instead.

**Comment by rnowosielski** [comment]
Can't we just write the code in a non-recursive manner ?

**Comment by mmaruniak** [comment]
Yes, I just added the #48 tag to this commit because it's related
